From: John Linville <linville@redhat.com>
Date: Mon, 7 Dec 2009 18:52:00 -0500
Subject: [net] mac80211: avoid uninit ptr deref in ieee80211
Message-id: <20091207185200.GA7022@redhat.com>
Patchwork-id: 21721
O-Subject: [RHEL5.5 PATCH] mac80211: avoid uninitialized ptr deref in
	ieee80211_change_iface
Bugzilla: 545121
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

ieee80211_change_iface dereferences an uninitialized pointer --
oops!  I missed this warning among some other warning SPAM (patch to
follow)... :-(

BZ545121

 net/mac80211/cfg.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Don Zickus <dzickus@redhat.com>

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 912b4b1..3ea9006 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -74,7 +74,7 @@ static int ieee80211_change_iface(struct wiphy *wiphy,
 	struct ieee80211_sub_if_data *sdata;
 	int ret;
 
-	if (netif_running(sdata->dev))
+	if (netif_running(dev))
 		return -EBUSY;
 
 	if (!nl80211_type_check(type))
